core/blockchain.go

```diff
@@ -2554,6 +2563,18 @@ func (bc *BlockChain) ValidatePayload(block *types.Block, feeRecipient common.Ad
 		return err
 	}
 
+	// Validate proposer payment
+
+	if !forceLastTxPayment {
```

Comment:
Hm, do we always expect the coinbase to be set to the proposer's fee recipient in this case? Maybe it'd be worth adding the check to make sure people aren't doing something more convoluted like internal transfers without consideration.
I'd be open to removing that check if people want us to, but I would be on the safer side if possible (either last tx payment or coinbase set to fee recipient)

Reply:
We also want to specifically handle case where payment is done in the last tx but it goes through some contract that can check block number and batch mev share kickbacks. 

Code that we have here handles both this case and coinbase case.

If we want to add restriction like "if coinbase is not the fee recipient than you have to pay with last tx but it also can be in internal transfer" this will make this code harder because you need to trace indivdual txs again (previous version of this pr) and it can be even more dangerous because of potential bugs and more complex logic.